ir-gen-cli-bundler/conjureircli/run.go

```diff
@@ -62,21 +62,21 @@ func InputPathToIR(inPath string) (rBytes []byte, rErr error) {
 }
 
 func InputPathToIRWithParams(inPath string, params ...Param) (rBytes []byte, rErr error) {
-	tmpDir, err := ioutil.TempDir("", "")
+	file, err := os.CreateTemp("", "")
```

Comment:
Per the [documentation](https://pkg.go.dev/os#CreateTemp):

> CreateTemp creates a new temporary file in the directory dir, opens the file for reading and writing, and returns the resulting file.

I believe we took the previous approach because `RunWithParams` takes a path to the output, which it is expected to create.

The proposed change here might work, but conceptually it's not great to create a new file that's open for read/write only to have another function overwrite that file.

We can add documentation to explain why we take the approach that we do, but given that, I would prefer to keep the current approach over making this change.